core/report/stdoutJson.go

```diff
@@ -54,12 +57,57 @@ func (s *stdoutJson) Start(input chan *types.Response) {
 	s.doneChan <- struct{}{}
 }
 
+type jsonResult struct {
+	SuccessCount int64                             `json:"success_count"`
+	FailedCount  int64                             `json:"fail_count"`
+	AvgDuration  float32                           `json:"avg_duration"`
+	ItemReports  map[int16]*jsonScenarioItemReport `json:"steps"`
+}
+
+type jsonScenarioItemReport struct {
+	Name           string             `json:"name"`
+	StatusCodeDist map[int]int        `json:"status_code_dist"`
+	ErrorDist      map[string]int     `json:"error_dist"`
+	Durations      map[string]float32 `json:"durations"`
+	Percentiles    map[string]float32 `json:"percentiles"`
+	SuccessCount   int64              `json:"success_count"`
+	FailedCount    int64              `json:"fail_count"`
+}
+
 func (s *stdoutJson) Report() {
-	p := 1e3
+	jsonResult := jsonResult{
+		SuccessCount: s.result.SuccessCount,
+		FailedCount:  s.result.FailedCount,
+		AvgDuration:  s.result.AvgDuration,
+		ItemReports:  map[int16]*jsonScenarioItemReport{},
+	}
+
+	for key, item := range s.result.ItemReports {
+		jsonResult.ItemReports[key] = &jsonScenarioItemReport{
+			Name:           item.Name,
+			StatusCodeDist: item.StatusCodeDist,
+			ErrorDist:      item.ErrorDist,
+			Durations:      item.Durations,
+			SuccessCount:   item.SuccessCount,
+			FailedCount:    item.FailedCount,
+		}
+
+		if s.reportPercentiles {
```

Comment:
If possible, let's not show the percentile key on if it is not enabled. 

Current state:
```
...
 "durations": {
        "connection": 0.005,
        "dns": 0.009,
        "request_write": 0,
        "response_read": 0.382,
        "server_processing": 0.253,
        "tls": 0.012,
        "total": 0.661
      },
      "percentiles": null, ---> Let's remove this if we can
      "success_count": 10,
      "fail_count": 0
    },
...
```